main.py

```diff
@@ -37,7 +37,7 @@
   
     chatbot_openai = ChatOpenAI()
 
-    slack_adapter = SlackAdapter(slack_app, chatbot_openai)
+    slack_adapter = SlackAdapter(slack_app, chatbot_openai, bot_controller)

```

Comment:
This might be better if we use the `bot_repository` directly since what we really need is the data access to the bots.